spanner/transaction_test.go

```diff
@@ -302,6 +302,186 @@ func TestReadWriteTransaction_ErrorReturned(t *testing.T) {
 	}
 }
 
+func TestClient_ReadWriteTransaction_UnimplementedErrorWithMultiplexedSessionSwitchesToRegular(t *testing.T) {
```

Comment:
Can you add a test showing that for other unimplmeneted error, readwrite txn still uses multiplexed sessions?
Ref - https://github.com/googleapis/java-spanner/blob/6cbaf7ec6502d04fc0a0c09720e2054bd10bead9/google-cloud-spanner/src/test/java/com/google/cloud/spanner/MultiplexedSessionDatabaseClientMockServerTest.java#L1520

Reply:
Updated this test